packages/block-editor/src/components/link-control/index.js

```diff
@@ -315,28 +495,56 @@ function LinkControl( {
 			ref={ wrapperNode }
 			className="block-editor-link-control"
 		>
-			{ isEditingLink || ! value ? (
+			{ isResolvingLink && (
+				<div className="block-editor-link-control__loading">
+					<Spinner /> { __( 'Creating' ) }…
+				</div>
+			) }
+
+			{ ( isEditingLink || ! value ) && ! isResolvingLink && (
 				<LinkControlSearchInput
 					value={ inputValue }
 					onChange={ onInputChange }
-					onSelect={ ( suggestion ) => {
-						onChange( { ...value, ...suggestion } );
-						stopEditing();
+					onSelect={ async ( suggestion ) => {
+						if (
+							suggestion.type &&
```

Comment:
This part of the condition is redundant, since it is never the case that `suggestion.type` would be both falsey and equal to the value of `CREATE_TYPE`.